datasources/src/test/scala/com/normation/plugins/datasources/UpdateHttpDatasetTest.scala

```diff
@@ -1197,42 +1175,36 @@ class UpdateHttpDatasetTest extends Specification with BoxSpecMatcher with Logga
     def test404prop(propName: String, initValue: Option[String], onMissing: MissingNodeBehavior, expectMod: Boolean)(
         finalStateCond: PROPS => MatchResult[PROPS]
     ): MatchResult[Any] = {
-      val (updates, infos) = buildNodeRepo(NodeConfigData.allNodeFacts)
+      val initNodeFacts = initValue match {
+        case None =>  NodeConfigData.allNodeFacts
+        case Some(init) => NodeConfigData.allNodeFacts.view.mapValues(n => n.copy(properties = n.properties :+ NodeProperty.apply(propName, init.toConfigValue, None, None))).toMap
+      }
+      val (updates, infos) = buildNodeRepo(initNodeFacts)
       val http             =
         new HttpQueryDataSourceService(infos, parameterRepo, interpolation, noPostHook, () => alwaysEnforce.succeed)
       val datasource       = NewDataSource(propName, url = s"${REST_SERVER_URL}/404", path = "$.some.prop", onMissing = onMissing)
 
       val nodes = infos.getAll()(QueryContext.testQC).toBox.openOrThrowException("test shall not throw")
       // set a value for all propName if asked
       val modId = ModificationId("set-test-404")
-      nodes.values.foreach { node =>
-        val newProps = CompareProperties
-          .updateProperties(
-            node.properties.toList,
-            Some(List(NodeProperty.apply(propName, initValue.getOrElse("").toConfigValue, None, None)))
-          )
-          .toBox
-          .openOrThrowException("test must be able to set prop")
-        val up       = node.modify(_.properties).setTo(Chunk.fromIterable(newProps))
-        infos.save(up)(ChangeContext(modId, actor, DateTime.now(), None, None, NodeSecurityContext.All)).runNow
```

Comment:
This runNow maybe ran after the emptying of the map below due to ZIO way of running things